pkg/ebpf/tracee.go

```diff
@@ -1108,55 +1109,38 @@ func (t *Tracee) getCapturedIfaceIdx(ifaceName string) (int, bool) {
 	return t.config.Capture.NetIfaces.Find(ifaceName)
 }
 
-const (
-	IoctlFetchSyscalls int32 = 1 << iota
-	IoctlHookedSeqOps
-)
+func (t *Tracee) triggerSyscallsIntegrityCheck() {
+	_, ok := t.events[events.HookedSyscalls]
+	if !ok {
+		return
+	}
+	t.triggerSyscallsIntegrityCheckCall()
+}
 
-// Struct names for the interfaces HookedSeqOpsEventID checks for hooks
-// The show,start,next and stop operation function pointers will be checked for each of those
-var netSeqOps = [6]string{
-	"tcp4_seq_ops",
-	"tcp6_seq_ops",
-	"udp_seq_ops",
-	"udp6_seq_ops",
-	"raw_seq_ops",
-	"raw6_seq_ops",
+// triggerSyscallsIntegrityCheck is used by a Uprobe to trigger an eBPF program that prints the syscall table
+//go:noinline
+func (t *Tracee) triggerSyscallsIntegrityCheckCall() {
 }
 
-func (t *Tracee) invokeIoctlTriggeredEvents(cmds int32) error {
-	// invoke HookedSyscallsEvent
-	if cmds&IoctlFetchSyscalls == IoctlFetchSyscalls {
-		_, ok := t.events[events.HookedSyscalls]
-		if ok {
-			ptmx, err := os.OpenFile(t.config.Capture.OutputPath, os.O_RDONLY, 0444)
-			if err != nil {
-				return err
-			}
-			syscall.Syscall(syscall.SYS_IOCTL, ptmx.Fd(), uintptr(IoctlFetchSyscalls), 0)
-			ptmx.Close()
-		}
+func (t *Tracee) triggerSeqOpsIntegrityCheck() {
+	_, ok := t.events[events.HookedSeqOps]
+	if !ok {
+		return
 	}
-
-	// invoke HookedSeqOps
-	if cmds&IoctlHookedSeqOps == IoctlHookedSeqOps {
-		_, ok := t.events[events.HookedSeqOps]
-		if ok {
-			ptmx, err := os.OpenFile(t.config.Capture.OutputPath, os.O_RDONLY, 0444)
-			if err != nil {
-				return err
-			}
-
-			for _, seq_name := range netSeqOps {
-				seqOpsStruct, err := t.kernelSymbols.GetSymbolByName("system", seq_name)
-				if err != nil {
-					continue
-				}
-				syscall.Syscall(syscall.SYS_IOCTL, ptmx.Fd(), uintptr(IoctlHookedSeqOps), uintptr(seqOpsStruct.Address))
-			}
-			ptmx.Close()
+	var seqOpsPointers [len(derive.NetSeqOps)]uint64
+	for i, seqName := range derive.NetSeqOps {
+		seqOpsStruct, err := t.kernelSymbols.GetSymbolByName("system", seqName)
+		if err != nil {
+			continue
 		}
+		seqOpsPointers[i] = seqOpsStruct.Address
 	}
+	t.triggerSeqOpsIntegrityCheckCall(seqOpsPointers)
+}
+
+// triggerSeqOpsIntegrityCheck is used by a Uprobe to trigger an eBPF program that prints the seq ops pointers
+//go:noinline
+func (t *Tracee) triggerSeqOpsIntegrityCheckCall(seqOpsStruct [len(derive.NetSeqOps)]uint64) error {
```

Comment:
Main challenge for moving the triggers might be defining the "probe arguments" more generically, but we'll tackle that when refactoring this.

Reply:
https://github.com/aquasecurity/tracee/issues/2055